### DISCUSSION TITLE
Creating a spectral image

### DISCUSSION THREAD

**Discussion by JoeD4321** [ImageMagick/ImageMagick#7549] [Help] [answered]
### ImageMagick version

7.1.1-33

### Operating system, version and so on

Fedora 40, 6.9.11-200.fc40.x86_64

### Description

Hello All,
I am trying to create an image that shows the full range of visible light, starting with RED on the left and VIOLET on the right.  I have no idea how to do this so I asked ChatGPT and got this:

`convert -size 200x200 gradient:'#FF0000-#FF7F00-#FFFF00-#00FF00-#0000FF-#4B0082-#8B00FF' spectral.png`

Other than changing **convert** to **magick** I ran as is.  It completed successfully, but the output image started with red on top and was just getting into orange at the bottom, avoiding all the other colors.  I think I am just missing another command qualifier but I can't figure it out.  Checked around and the search engines with no luck.

Can someone give me the correct command?

**Comment by fmw42** [comment]
See for example http://www.fmwconcepts.com/imagemagick/tidbits/graphics.php#rainbow

**Comment by JoeD4321** [comment, marked as answer]
I used the examples suggested above and built commands that do what I need.

`magick xc:red xc:orange xc:yellow xc:green1 xc:cyan xc:blue xc:blueviolet +append -filter Cubic -resize 200x200! spectral.png
`

I also wanted the same thing in reverse VIBGYOR rather than ROYGBIV so I used this:
`magick xc:blueviolet xc:blue xc:cyan xc:green1 xc:yellow xc:orange xc:red +append -filter Cubic -resize 200x200! spectral2.png
`

Files attached
![spectral](https://github.com/user-attachments/assets/94a54d75-2827-4a78-8e75-061220d7ac27)
![spectral2](https://github.com/user-attachments/assets/655374c9-2db0-42d5-8b68-72b7ee0a358b)

**Reply by fmw42** [reply]
You could have used the first one and just added -flop to reverse horizontally.
